client/sync.js

```diff
@@ -47,68 +47,8 @@ const logSync = (message, logLevel = DEBUG) => {
   }
 }
 
-/**
- * decrypt then deserialize a message.
- * @param {Uint8Array} ciphertext
- * @returns {Object}
- */
-const decrypt = (ciphertext) => {
-  const d = clientSerializer.byteArrayToSecretboxRecord(ciphertext)
-  const decrypted = crypto.decrypt(d.encryptedData,
-    d.nonceRandom, clientKeys.secretboxKey)
-  if (!decrypted) {
-    throw new Error('Decryption failed.')
-  }
-  return clientSerializer.byteArrayToSyncRecord(decrypted)
-}
-
-/**
- * serialize then encrypts a sync record
- * @param {Object} message
- * @returns {Uint8Array}
- */
-const encrypt = (message) => {
-  const s = clientSerializer.syncRecordToByteArray(message)
-  const nonceRandom = crypto.randomBytes(20)
-  const encrypted = crypto.encrypt(s, clientKeys.secretboxKey,
-    conf.nonceCounter, nonceRandom)
-  return clientSerializer.SecretboxRecordToByteArray({
-    nonceRandom,
-    counter: conf.counter,
-    encryptedData: encrypted.ciphertext
-  })
-}
-
-/**
- * Gets AWS creds.
- * @returns {Promise}
- */
-const getAWSCredentials = () => {
-  const serverUrl = config.serverUrl
-  const now = Math.floor(Date.now() / 1000).toString()
-  if (clientSerializer === null) {
-    throw new Error('Serializer not initialized.')
-  }
-  const userId = window.encodeURIComponent(clientUserId)
-  const request = new window.Request(`${serverUrl}/${userId}/credentials`, {
-    method: 'POST',
-    body: crypto.sign(clientSerializer.stringToByteArray(now), clientKeys.secretKey)
-  })
-  return window.fetch(request)
-    .then((response) => {
-      if (!response.ok) {
-        throw new Error('Credential server response ' + response.status)
-      }
-      return response.arrayBuffer()
-    })
-    .then((buffer) => {
-      requester = new RequestUtil(clientSerializer, new Uint8Array(buffer),
-        config.apiVersion, clientUserId)
-      if (!requester.s3) {
-        throw new Error('could not initialize AWS SDK')
-      }
-    })
-}
+const decrypt = cryptoUtil.Decrypt(clientSerializer, clientKeys.secretboxKey)
+const encrypt = cryptoUtil.Encrypt(clientSerializer, clientKeys.secretboxKey, conf.nonceCounter)
```

Comment:
these functions need to be initialized after clientSerializer is initialized, otherwise clientSerializer is null.